src/js/components/lazy-load.js

```diff
@@ -11,49 +9,116 @@ const DEFAULTS = {
 }
 
 class LazyLoad {
+  /**
+   * Create a lazy load component
+   * Set base methods references and props (element and options)
+   *
+   * @param  {[DOM]} element
+   * @param  {[Object]} options
+   */
   constructor (element, options) {
-    this.$element = $(element)
-    this.options = $.extend({}, DEFAULTS, (options || {}))
+    this.element = element
+    this.options = { ...DEFAULTS, ...options }
+
+    this.onScrollHandler = this.onScrollHandler.bind(this)
+    this.onScroll = this.onScroll.bind(this)
+    this.checkPlaceholder = this.checkPlaceholder.bind(this)
   }
 
+  /**
+   * Set event listeners and check placeholder visibility
+   *
+   * @return {[Class]}
+   */
   init () {
     this.bindListeners()
     this.checkVisiblePlaceholders()
 
     return this
   }
 
+  /**
+   * Add scroll event to window and attach @onScrollHandler to event
+   */
   bindListeners () {
-    this.onScrollHandler = throttle(this.onScroll.bind(this), this.options.throttle)
-    $(window).on('scroll', this.onScrollHandler)
+    window.addEventListener('scroll', this.onScrollHandler)
+  }
+
+  /**
+   * Prevent @onScroll to be called multiple times using throttle
+   */
+  onScrollHandler () {
+    throttle(this.onScroll, this.options.throttle)
   }
 
+  /**
+   * Check if exist at least on [data-lazy] and check visibility
+   * If don't have any [data-lazy] remove scroll event to prevent
+   * handler been called unnecessarily
+   */
   onScroll () {
-    return this.$element.length ? this.checkVisiblePlaceholders() : $(window).off('scroll', this.onScrollHandler)
+    if (this.element.length) {
+      return this.checkVisiblePlaceholders()
+    }
+
+    return window.removeEventListener('scroll', this.onScrollHandler)
   }
 
+  /**
+   * Set body offsetHeight and offsetWidth as props
+   * and check placeholder visibility
+   */
   checkVisiblePlaceholders () {
-    this.windowHeight = $(window).height()
-    this.windowWidth = $(window).width()
+    const { offsetHeight, offsetWidth } = document.body
 
-    Array.prototype.forEach.call(this.$element, this.checkPlaceholder.bind(this))
+    this.windowHeight = offsetHeight
+    this.windowWidth = offsetWidth
+
+    Array.from(this.element).forEach(this.checkPlaceholder)
   }
 
+  /**
+   * Check if placeholder is visible, if it's true render imagem
+   * and remove this element from this.element Array
+   *
+   * @param  {[DOM]} placeholder
+   */
   checkPlaceholder (placeholder) {
     if (this.isPlaceholderVisible(placeholder)) {
       this.renderImage(placeholder)
-      removeFromArray(this.$element, placeholder)
+      removeFromArray(this.element, placeholder)
     }
   }
 
+  /**
+   * Check if placeholder position in window
```

Comment:
`Check if placeholder is currently visible by comparing its position with the window position.`